drivers/input/input_cy8cmbr3xxx.c

```diff
@@ -0,0 +1,451 @@
+#define CY8CMBR3XXX_DEBUG_CP                  0xDD
+#define CY8CMBR3XXX_DEBUG_DIFFERENCE_COUNT0   0xDE
+#define CY8CMBR3XXX_DEBUG_BASELINE0           0xE0
+#define CY8CMBR3XXX_DEBUG_RAW_COUNT0          0xE2
+#define CY8CMBR3XXX_DEBUG_AVG_RAW_COUNT0      0xE4
+#define CY8CMBR3XXX_SYNC_COUNTER2             0xE7
+
+/* The controller wakes up from the low-power state on an address match but sends NACK
+ * until it transitions into the Active state. When the device NACKs a transaction, the host is
+ * expected to retry the transaction until it receives an ACK. Typically, no more than 3 retries are
+ * necessary, depending on time between the interrupt and the first i2c transfer or if
+ * no interrupt has caused the initiation of the communication.
+ */
+#define CY8CMBR3XXX_I2C_RETRIES 5
+
+struct cy8cmbr3xxx_config {
+	struct i2c_dt_spec i2c;
+	struct gpio_dt_spec int_gpio;
+	struct gpio_dt_spec rst_gpio;
+	const uint16_t *input_codes;
+	uint8_t input_codes_count;
+	const uint16_t *proximity_codes;
+	uint8_t proximity_codes_count;
+};
+
+struct cy8cmbr3xxx_data {
+	const struct device *dev;
+	struct k_work work;
+	struct gpio_callback int_gpio_cb;
+	uint16_t prev_button_state;
+	uint8_t prev_proximity_state;
+};
+
+static int cypress_cy8cmbr3xxx_i2c_read(const struct device *dev, uint8_t address, void *buf,
+					size_t len)
+{
+	const struct cy8cmbr3xxx_config *config = dev->config;
+	int ret;
+
+	for (int i = 0; i < CY8CMBR3XXX_I2C_RETRIES; i++) {
+		ret = i2c_write_read_dt(&config->i2c, &address, sizeof(address), buf, len);
+		if (ret == 0) {
+			break;
+		}
+	}
+
+	return ret;
+}
+
+static int cypress_cy8cmbr3xxx_i2c_write(const struct device *dev, uint8_t address, const void *buf,
+					 size_t len)
+{
+	const struct cy8cmbr3xxx_config *config = dev->config;
+	int ret;
+
+	for (int i = 0; i < CY8CMBR3XXX_I2C_RETRIES; i++) {
+		ret = i2c_burst_write_dt(&config->i2c, address, buf, len);
+		if (ret == 0) {
+			break;
+		}
+	}
+
+	return ret;
+}
+
+static int cypress_cy8cmbr3xxx_wait_for_command_completion(const struct device *dev,
+							   k_timeout_t timeout)
+{
+	uint8_t current_command;
+	int ret;
+	k_timepoint_t end = sys_timepoint_calc(timeout);
+
+	do {
+		/* Wait for the completion of the command. After a reset command, it can
+		 * happen that the device NACKs for some time.
+		 */
+		ret = cypress_cy8cmbr3xxx_i2c_read(dev, CY8CMBR3XXX_CTRL_CMD, &current_command,
+						   sizeof(uint8_t));
+
+		/* As soon as current_command is 0x00, the command is completed */
+		if (ret == 0 && current_command == 0x00) {
+			break;
+		}
+
+		if (sys_timepoint_expired(end)) {
+			LOG_ERR("Wait for command completion timed out");
+			return -ETIMEDOUT;
+		}
+
+		k_msleep(1);
+	} while (true);
+
+	return 0;
+}
+
+int cypress_cy8cmbr3xxx_configure(const struct device *dev,
+				  const struct cy8cmbr3xxx_config_data *config)
+{
+	int ret;
+	uint8_t read_config[CY8CMBR3XXX_EZ_CLICK_CONFIG_SIZE] = {0};
+
+	if (config == NULL) {
+		LOG_ERR("Config is NULL");
```

Comment:
nit: I'd drop the log, save few bytes of flash, this is very unlikely to happen anyway